src/Geta.NotFoundHandler.Admin/Areas/GetaNotFoundHandlerAdmin/Pages/Index.cshtml.cs

```diff
@@ -58,10 +64,88 @@ public IActionResult OnPostCreate()
         return RedirectToPage();
     }
 
-    public IActionResult OnPostDelete(string oldUrl)
+    public IActionResult OnPostDelete(RedirectsRequest request)
     {
-        _redirectsService.DeleteByOldUrl(oldUrl);
-        return RedirectToPage();
+        ModelState.Clear();
+
+        _redirectsService.DeleteById(request.Id);
+        
+        ApplyRequest(request);
+
+        Load();
+        
+        return RedirectToPage(request);
+    }
+
+    public IActionResult OnPostEdit(RedirectsRequest request)
+    {
+        ModelState.Clear();
+
+        ApplyRequest(request);
+
+        Load();
+
+        var redirect = _redirectsService.Get(request.Id);
+
+        if (redirect != null)
+        {
+            ActivateEditMode(request.Id);
+
+            CustomRedirect = new RedirectModel
+            {
+                OldUrl = redirect.OldUrl,
+                RedirectType = redirect.RedirectType,
+                NewUrl = redirect.NewUrl,
+                WildCardSkipAppend = redirect.WildCardSkipAppend
+            };
+        }
+
+        return Page();
+    }
+    
+    public IActionResult OnPostUpdate(RedirectsRequest request)
+    {
+        if (!ModelState.IsValid)
+        {
+            ApplyRequest(request);
+
+            Load();
+
+            ActivateEditMode(request.Id);
+
+            return Page();
+        }
+
+        _redirectsService.AddOrUpdate(new CustomRedirect
+        {
+            Id = request.Id,
+            OldUrl = CustomRedirect.OldUrl,
+            RedirectType = CustomRedirect.RedirectType,
+            NewUrl = CustomRedirect.NewUrl,
+            WildCardSkipAppend = CustomRedirect.WildCardSkipAppend
+        });
+
+        return RedirectToPage(request);
+    }
+    
+    public IActionResult OnPostCancelEdit(RedirectsRequest request)
+    {
+        return RedirectToPage(request);
```

Comment:
I don't think this will remove id from the query string.

Reply:
Not used anymore